lib/iris/tests/integration/test_grib2.py

```diff
@@ -36,6 +36,20 @@
 
 if tests.GRIB_AVAILABLE:
     import gribapi
+    from iris.fileformats.grib._message import _GribMessage
+
+
+def _evaluate_grib_message(test, filename, contents):
+    """
+    Evaluate whether all messages in a GRIB2 file contains contents:
+    an iterable of keys and values.
+
+    """
+    messages = _GribMessage.messages_from_filename(filename)
+    for message in messages:
+        for element in contents:
+            section, key, val = element
+            test.assertEqual(message.sections[section][key], val)
```

Comment:
@marqh I was initially confused as to what exactly the `test` argument was ... but from the usage it's clear that it's an instance of a test class, for a particular test case.

My only suggestion to avoid this confusion would be to do this:
```python
class _Test(tests.IrisTest):
    def _evaluate_grib_message(self, filename, contents):
        ....

@tests.skip_data
@tests.skip_grib
class TestPDT11(_Test):
    ...
    def test_perturbation(self):
        ....
        self._evaluate_grib_message(temp_file_path, expect_values)

@tests.skip_data
@tests.skip_grib
class TestGDT5(_Test):
    ....
```

If you bought into this alternative approach, then I'd also attempt to sell you on changing the method name from `_evaluate_grib_message` to `assertGribMessage` ... infact, it might be worth having the discussion as to whether `assertGribMessage` should be elevated to be a method on `iris.tests.IrisTest` rather than on a private class; I guess it depends on the scope of its intended usage ... other devs might have an opinion on this, *nudge* @rhattersley